### DISCUSSION TITLE
Math-Functions

### DISCUSSION THREAD

**Discussion by chrgweber** [pybricks/support#1596] [Q&A]
Hello together,
I am using pybricks micropython in the Version 2.0.0. Is it possible to use built-in math function like tan, sin or cos.
I was not able to call math.tan or umath.sin. Neither I was able to import a lib.
Thanks for you help.
Regards
Chris

**Comment by laurensvalk** [comment]
What error did you get when trying to import the math library?

**Reply by chrgweber** [reply]
Hi Laurens,
it turns out, that the intellisense function in Visual Studio Code does not show up, when I code "from math import tan, pi".
So I thought tan or pi are not available.

Now, when I use tan or pi later in the code, everything works fine.
Kind regards
Chris

**Reply by laurensvalk** [reply]
Thanks, that's good to know. Those were probably not yet documented in the 2.0.0 version. 

Although EV3 is still at 2.0.0, the math functions haven't changed much since, so you could use [this newer page](https://docs.pybricks.com/en/stable/micropython/umath.html) as a reference.
